redpanda/resources/serverlesscluster/data_cluster.go

```diff
@@ -0,0 +1,129 @@
+func (*DataSourceServerlessCluster) Metadata(_ context.Context, _ datasource.MetadataRequest, response *datasource.MetadataResponse) {
+	response.TypeName = "redpanda_serverless_cluster"
+}
+
+// Configure uses provider level data to configure DataSourceServerlessCluster's client.
+func (d *DataSourceServerlessCluster) Configure(_ context.Context, req datasource.ConfigureRequest, resp *datasource.ConfigureResponse) {
+	if req.ProviderData == nil {
+		return
+	}
+
+	p, ok := req.ProviderData.(config.Datasource)
+
+	if !ok {
+		resp.Diagnostics.AddError(
+			"Unexpected Resource Configure Type",
+			fmt.Sprintf("Expected *provider.Data, got: %T. Please report this issue to the provider developers.", req.ProviderData))
+		return
+	}
+	d.CpCl = cloud.NewControlPlaneClientSet(p.ControlPlaneConnection)
+}
+
+// Read reads the ServerlessCluster data source's values and updates the state.
+func (d *DataSourceServerlessCluster) Read(ctx context.Context, req datasource.ReadRequest, resp *datasource.ReadResponse) {
+	var model models.ServerlessCluster
+	resp.Diagnostics.Append(req.Config.Get(ctx, &model)...)
+
+	serverlessCluster, err := d.CpCl.ServerlessClusterForID(ctx, model.ID.ValueString())
+	if err != nil {
+		if utils.IsNotFound(err) {
+			resp.Diagnostics.AddError(fmt.Sprintf("unable to find serverless cluster %s", model.ID), err.Error())
+			return
+		}
+		resp.Diagnostics.AddError(fmt.Sprintf("failed to read serverless cluster %s", model.ID), err.Error())
+		return
+	}
+	clusterURL, err := utils.SplitSchemeDefPort(serverlessCluster.DataplaneApi.Url, "443")
+	if err != nil {
+		resp.Diagnostics.AddError("unable to parse Cluster API URL", err.Error())
+		return
+	}
+	// Mapping the fields from the serverless cluster to the Terraform state
+	resp.Diagnostics.Append(resp.State.Set(ctx, &models.ServerlessCluster{
+		Name:             types.StringValue(serverlessCluster.Name),
+		ServerlessRegion: types.StringValue(serverlessCluster.ServerlessRegion),
+		ResourceGroupID:  types.StringValue(serverlessCluster.ResourceGroupId),
+		ID:               types.StringValue(serverlessCluster.Id),
+		ClusterAPIURL:    types.StringValue(clusterURL),
+	})...)
+}
+
+// Schema returns the schema for the ServerlessCluster data source.
+func (*DataSourceServerlessCluster) Schema(_ context.Context, _ datasource.SchemaRequest, resp *datasource.SchemaResponse) {
+	resp.Schema = datasourceServerlessClusterSchema() // Reuse the schema from the resource
+}
+
+func datasourceServerlessClusterSchema() schema.Schema {
+	return schema.Schema{
+		Attributes: map[string]schema.Attribute{
+			"id": schema.StringAttribute{
+				Required:    true,
+				Description: "The ID of the serverless cluster",
+			},
+			"name": schema.StringAttribute{
+				Computed:    true,
+				Description: "Name of the serverless cluster",
+			},
+			"serverless_region": schema.StringAttribute{
+				Computed:    true,
+				Description: "Cloud provider specific region of the serverless cluster",
```

Comment:
It might be good to extend this with a doc link as I personally wasn't able to find a list of serverless regions with a quick google.